### PR TITLE
Fix log message format bugs

### PR DESCRIPTION
**Pull request by joegallo**
The codebase is very trappy on this, which is unfortunate, but at present we have no way of preventing this from happening (at least in the broadest sense, I think some simple special cases are checked automatically) besides being really careful all the time.

I just happened to notice that this snuck in via #117575, so I'm chasing that PR (hence the wide list of backports). I'm also including @original-brownbear as a reviewer because of the snapshot code.

### CODE REVIEW

**Review by nielsbauman — APPROVED** [review]
Thanks, Joe!

**Comment by joegallo** [comment]
Related to https://github.com/elastic/elasticsearch/issues/104782, in that with the changing of the seasons, I sometimes get the urge to ~fly south for the winter~ `grep` the codebase for these problems. But maybe we should 'just' have the build fail if problems like this sneak in?

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java [hunk 1]
```diff
@@ -3485,7 +3485,7 @@ private static void ensureNotAborted(ShardId shardId, SnapshotId snapshotId, Ind
                 // the ensureNotAborted() call above did not throw. The remaining options don't make sense, if they ever happen.
                 logger.error(
                     () -> Strings.format(
```
Comment (original-brownbear) [inline server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java:3487]: This one is kinda nonsense? :) Why use a lambda and String.format here when you're not logging an exception?
I know it was like that before but maybe just remove the format call and lambda? :)

**Review by original-brownbear — APPROVED** [review]
LGTM one cleanup suggestion though :)

File: server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java (hunk 1 above)
Comment (joegallo) [reply server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java]: Sure thing, 6a21cc98b3ff7403cce6a033b1d7a0f88f87c56d.

File: qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/DownsampleIT.java [hunk 2]
```diff
@@ -238,7 +238,7 @@ private String getRollupIndexName() throws IOException {
         if (asMap.size() == 1) {
             return (String) asMap.keySet().toArray()[0];
         }
-        logger.warn("--> No matching rollup name for path [%s]", endpoint);
```
Comment (samxbr) [inline qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/DownsampleIT.java:241]: Just dropping by, typo here  `{}}` -> `{}`?

File: qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/DownsampleIT.java [hunk 3]
```diff
@@ -238,7 +238,7 @@ private String getRollupIndexName() throws IOException {
...
+        logger.warn("--> No matching rollup name for path [{}}]", endpoint);
```
Comment (samxbr) [inline qa/rolling-upgrade/src/javaRestTest/java/org/elasticsearch/upgrades/DownsampleIT.java:241]: Just dropping by, typo here  `{}}` -> `{}`?

**Review by samxbr — APPROVED** [review]
LGTM

**Pull request merged**